patterns/graphs.md

```diff
@@ -0,0 +1,97 @@
+# Graphs
+
+## Description
+
+Implement a graph data structure that is dynamically modifiable during runtime.
+
+
+## Example
+
+```rust
+// A node of the graph.
+pub struct Node<T> {
+    parent: Option<NodeId>,
```

Comment:
If a node can have at most one parent, then the data structure is a *tree*, not a graph. Nodes in a graph can have more than one parent.

Reply:
It looks more like a *forest*, since this code allows for multiple roots 😉